documents/extension/funding/service.go

```diff
@@ -0,0 +1,178 @@
+package funding
+
+import (
+	"context"
+	"reflect"
+	"strconv"
+	"strings"
+
+	"github.com/centrifuge/go-centrifuge/utils"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+
+	"github.com/centrifuge/go-centrifuge/centerrors"
+	"github.com/centrifuge/go-centrifuge/documents"
+	"github.com/centrifuge/go-centrifuge/errors"
+	clientfundingpb "github.com/centrifuge/go-centrifuge/protobufs/gen/go/funding"
+)
+
+// Service defines specific functions for extension funding
+type Service interface {
+	documents.Service
+	// DeriveFromPayload derives Funding from clientPayload
+	DeriveFromPayload(ctx context.Context, req *clientfundingpb.FundingCreatePayload, identifier []byte) (documents.Model, error)
+
+	// DeriveFundingResponse returns a funding in client format
+	DeriveFundingResponse(model documents.Model, payload *clientfundingpb.FundingCreatePayload) (*clientfundingpb.FundingResponse, error)
+}
+
+// service implements Service and handles all funding related persistence and validations
+type service struct {
+	documents.Service
+	tokenRegFinder func() documents.TokenRegistry
+}
+
+const fundingLabel = "centrifuge_funding"
```

Comment:
do we need to say `centrifuge` here?

Reply:
mhm. For the other types, we are using the protobuf github link. Like invoice etc.

Reply:
we could just say `funding` ?